registry/handlers/api_test.go

```diff
@@ -2242,7 +2268,12 @@ func finishUpload(t *testing.T, ub *v2.URLBuilder, name reference.Named, uploadU
 	return resp.Header.Get("Location")
 }
 
-func doPushChunk(t *testing.T, uploadURLBase string, body io.Reader) (*http.Response, digest.Digest, error) {
+type chunkOption struct {
+	// Content-Range
+	cr string
+}
+
+func doPushChunk(t *testing.T, uploadURLBase string, body io.Reader, option ...chunkOption) (*http.Response, error) {
```

Comment:
I'm guessing you're using `...chunkOption` to make the signature backward compatible. 

However, this function is not exported, and is only able to handle the first option passed (if multiple options are passed, they will be ignored).

If you took this approach to allow future extensions; given that `chunkOption` is a struct, that would already be possible by adding additional properties to `chunkOption`.

I think better options would be to either;

Accept a single option;

```suggestion
func doPushChunk(t *testing.T, uploadURLBase string, body io.Reader, options chunkOptions) (*http.Response, error) {
```

Or, if we don't anticipate further options to be added in the near future, keep it simple, and pass the content-range directly. If further options are added in future, it can be rewritten to use a struct, or functional arguments)

```suggestion
func doPushChunk(t *testing.T, uploadURLBase string, body io.Reader, contentRange string) (*http.Response, error) {
```

Reply:
I pefer to reserve the options struct as it's easy to support the future extensions.